prolif/residue.py

```diff
@@ -186,7 +196,7 @@ class ResidueGroup(UserDict):
     access a subset of a ResidueGroup.
     """
 
-    def __init__(self, residues: list[Residue]):
+    def __init__(self, residues: list["Residue"]):
         self._residues = np.asarray(residues, dtype=object)
```

Comment:
```suggestion
        self._residues = cast(Sequence[Residue], np.asarray(residues, dtype=object))
```
It's not perfect but it fixes a lot of other issues 🤷